src/parser/flowchart.ts

```diff
@@ -228,9 +240,12 @@ export const parseMermaidFlowChartDiagram = (
   Object.keys(vertices).forEach((id) => {
     vertices[id] = parseVertex(vertices[id], containerEl);
   });
+
+  const processedEdges = new Set<string>();
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** ‏I feel there's a better way to do this but I haven't figured it out yet, the same happens in the state diagram.

Reply:
I have made into a function closure for better encapsulation

Reply:
I added a map with the aim of avoiding a for loop inside parseEdges, and the only way I thought to encapsulate it was to create a kind of factory haha.
```js
function createEdgeParser() {
  const edgeCountMap = new Map<string, number>();

  return function parseEdge(data: any, containerEl: HTMLElement) {
    const edgeIdentifier = `${data.start}-${data.end}`;
    const count = edgeCountMap.get(edgeIdentifier) || 0;
    edgeCountMap.set(edgeIdentifier, count + 1);

    // Rest of parseEdge implementation...

    return {

    };
  };
}